newIDE/app/src/ResourcesList/ResourceSelector.js

```diff
@@ -163,32 +163,74 @@ export default class ResourceSelector extends React.Component<Props, State> {
   };
 
   _editWith = (resourceExternalEditor: ResourceExternalEditor) => {
-    const { project, resourcesLoader } = this.props;
+     const {
+      project,
+      resourcesLoader,
+      initialResourceName,
+      resourceKind,
+    } = this.props;
     const { resourceName } = this.state;
-
-    const resourceNames = [];
-    if (project.getResourcesManager().hasResource(resourceName)) {
-      resourceNames.push(resourceName);
+    const resourcesManager = project.getResourcesManager();
+    const initialResource = resourcesManager.getResource(initialResourceName)
+    
+    let initialResourceMetadata = {};
+    const initialResourceMetadataRaw = initialResource.getMetadata();
+    if (initialResourceMetadataRaw) {
+      try {
+        initialResourceMetadata = JSON.parse(initialResourceMetadataRaw);
+      } catch(e) { console.error("Malformed metadata", e); }
+      console.log('resource metadata found, sending it to editor:');
+      console.log(initialResourceMetadata)
     }
 
-    resourceExternalEditor.edit({
-      project,
-      resourcesLoader,
-      singleFrame: true,
-      resourceNames,
-      extraOptions: {
-        fps: 0,
-        name: 'Image',
-        isLooping: false,
-      },
-      onChangesSaved: resources => {
-        if (!resources.length) return;
+    let initialResourcePath = resourcesLoader.getFullUrl(project, initialResourceName)
+    initialResourcePath = initialResourcePath.substring(
+      7,
+      initialResourcePath.lastIndexOf('?cache=')
+    );
 
-        // Burst the ResourcesLoader cache to force images to be reloaded (and not cached by the browser).
-        resourcesLoader.burstUrlsCacheForResources(project, [resources[0].name]);
-        this.props.onChange(resources[0].name);
-      },
-    });
+    let externalEditorOptions = null;
+    if (resourceKind === 'image') {
+      const resourceNames = [];
+      if (resourcesManager.hasResource(resourceName)) {
+        resourceNames.push(resourceName);
+      }
+      externalEditorOptions = {
+        project,
+        resourcesLoader,
+        singleFrame: true,
+        resourceNames,
+        extraOptions: {
+          fps: 0,
+          name: 'Image',
+          isLooping: false,
+        },
+        onChangesSaved: resources => {
+          if (!resources.length) return;
+
+          // Burst the ResourcesLoader cache to force images to be reloaded (and not cached by the browser).
+          resourcesLoader.burstUrlsCacheForResources(project, [
+            resources[0].name,
+          ]);
+          this.props.onChange(resources[0].name);
+        },
+      };
+    } else if (resourceKind === 'audio') {
+      externalEditorOptions = {
+        project,
+        extraOptions: {
+          initialResourceName,
```

Comment:
Are you sure you need `initialResourceName`?

Reply:
ah you're right. I just realized it is already declared in
`const { resourceName } = this.state;`